src/ui/interface.js

```diff
@@ -680,6 +669,47 @@ export class UI {
 		// Show UI
 		this.$display.show();
 		this.$dash.hide();
+
+		// Events
+		this.game.signals.ui.add(this._handleUiEvent, this);
+	}
+
+	/**
+	 * Handle events on the "ui" channel.
+	 *
+	 * @param {string} message Event name.
+	 * @param {object} payload Event payload.
+	 */
+	_handleUiEvent(message, payload) {
+		if (message === 'toggleDash') {
+			this.toggleDash();
+			this.closeMusicPlayer();
+			this.closeScoreboard();
+		}
+
+		if (message === 'toggleScore') {
+			this.toggleScoreboard();
+			this.closeDash();
+			this.closeMusicPlayer();
+		}
+
+		if (message === 'toggleMusicPlayer') {
+			this.toggleMusicPlayer();
+			this.closeDash();
+			this.closeScoreboard();
+		}
+
+		if (message === 'toggleMetaPowers') {
+			this.closeDash();
+			this.closeMusicPlayer();
+			this.closeScoreboard();
+		}
+
+		if (message === 'closeInterfaceScreens') {
+			this.closeDash();
+			this.closeMusicPlayer();
+			this.closeScoreboard();
+		}
```

Comment:
Hmm, this seems a bit repetitive. Could probably use a dictionary with screens and a function to take one parameter, remove that one from the dictionary then just close all others. Though we *might* just use something like Vue.js at some point if it makes sense in order to handle some of the UI (deprecate jQuery as well in the process) and also integrate website functionality into the game itself and have the game displayed first thing on the domain name.